### PR TITLE
feature: add automatic labeling of media, equations, and tables

### PR DESCRIPTION
**Pull request by 3mcd**
This PR adds automatic labeling of audio, video, images, equations, and tables. The labels aren't customizable yet, but this code adds a hook into the labeling logic via the `blockNames` object injected via document state.

### CODE REVIEW

File: client/components/Editor/styles/table.scss [hunk 1]
```diff
@@ -14,20 +24,17 @@ td {
 		margin-bottom: 0;
 	}
 }
+
 /* This clunky table, tableWrapper thing is because */
```
Comment (idreyn) [inline client/components/Editor/styles/table.scss:28]: This comment (or at least its specific wording) appears to be out of date following these changes.

**Review by idreyn — APPROVED** [review]
No major concerns here, and I like the `buildLabel` function!

As I mentioned on Slack I think the approach of using a `NodeView` to render reference nodes may be more straightforward, since you can read from `editorView.editable` to determine whether links should be followable or not. This lets us avoid using a reactive attribute on every reference node to hold `isReadOnly`, and has the benefit of actually responding appropriately when the editor's, uh, editability, changes. I'll let you make the call but I do think it's a more idiomatic design.

File: client/components/Editor/Editor.tsx [hunk 2]
```diff
@@ -122,6 +144,7 @@ const Editor = (props: Props) => {
 				handleClickOn: props.handleSingleClick,
 				handleDoubleClickOn: props.handleDoubleClick,
 				handleScrollToSelection: props.onScrollToSelection,
+				clipboardSerializer: wrapDomSerializer(DOMSerializer.fromSchema(schema)),
```
[the remark this answers is not in the document]
Comment (3mcd) [reply client/components/Editor/Editor.tsx:147]: Yep, sounds good!

File: client/components/Editor/styles/table.scss (hunk 1 above)
Comment (3mcd) [reply client/components/Editor/styles/table.scss:28]: Got it, I'll update it or remove it.

**Pull request merged**